src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -685,6 +685,110 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
                 self.fail("Callback was not called")
 
 
+class _LoggingServerInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, tag, record):
+        self.tag = tag
+        self.record = record
+
+    async def intercept_service(self, continuation, handler_call_details):
+        self.record.append(self.tag + ':intercept_service')
+        return await continuation(handler_call_details)
+
+
+class _GenericServerInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, fn):
+        self._fn = fn
+
+    async def intercept_service(self, continuation, handler_call_details):
+        return await self._fn(continuation, handler_call_details)
+
+
+def _filter_server_interceptor(condition, interceptor):
+    async def intercept_service(continuation, handler_call_details):
+        if condition(handler_call_details):
+            return await interceptor.intercept_service(continuation,
+                                                       handler_call_details)
+        return await continuation(handler_call_details)
+
+    return _GenericServerInterceptor(intercept_service)
+
+
+class TestServerInterceptor(AioTestBase):
+    async def setUp(self) -> None:
+        self._record = []
+        conditional_interceptor = _filter_server_interceptor(
+            lambda x: ('secret', '42') in x.invocation_metadata,
+            _LoggingServerInterceptor('log3', self._record))
+        self._interceptors = (
+            _LoggingServerInterceptor('log1', self._record),
+            conditional_interceptor,
+            _LoggingServerInterceptor('log2', self._record),
+        )
+        self._server_target, self._server = await start_test_server(
+            interceptors=self._interceptors)
+
+    async def tearDown(self) -> None:
+        self._server.stop(None)
+
+    async def test_invalid_interceptor(self):
+        class InvalidInterceptor:
+            """Just an invalid Interceptor"""
+
+        with self.assertRaises(aio.AioRpcError):
+            server_target, _ = await start_test_server(
+                interceptors=(InvalidInterceptor(),))
```

Comment:
a `Channel` with an invalid interceptor is not allowed to be instantiated [1 ], having the feeling that we would need to do the same with the server.

[1] https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/experimental/aio/_channel.py#L317